caddy.go

```diff
@@ -522,15 +520,18 @@ func finishSettingUp(ctx Context, cfg *Config) error {
 					case <-timer.C:
 						loadedConfig, err := val.(ConfigLoader).LoadConfig(ctx)
 						if err != nil {
-							logger.Error("failed loading dynamic config; will retry", zap.Error(err))
+							if errors.Is(err, ErrConfigUnchanged) {
+								logger.Info("dynamically-loaded config was unchanged; will retry")
+							} else {
+								logger.Error("failed loading dynamic config; will retry", zap.Error(err))
+							}
 							continue
 						}
 						if loadedConfig == nil {
 							logger.Info("dynamically-loaded config was nil; will retry")
 							continue
 						}
```

Comment:
It's been a while and we've been through several iterations now, so help me out... I'm trying to recall why simply returning `nil, nil` is insufficient for just retrying the load again later. Why do we need to return a sentinel error value at all?

Reply:
Ah, I remember noticing that there's a judgement of nil config, but somehow turned a blind eye to it later.

Considering that LoadConfig is a public API, if there is no available config, I would prefer it to return a sentinel error. Maybe this is just a matter of taste, or maybe this is more conventional (like [DB.QueryRow](https://pkg.go.dev/database/sql#DB.QueryRow) and https://github.com/99designs/gqlgen/discussions/1891)?

That being said, I think returning `nil, nil` is still a feasible solution to avoid producing misleading error messages I mentioned before https://github.com/caddyserver/caddy/pull/4615#issuecomment-1072004059. I think this is not a big deal, so it's also OK if this is the way Caddy prefers :)